src/io/framing.rs

```diff
@@ -72,9 +73,9 @@ impl<T, P, S> Transport for Framed<T, P, S>
           S: Serialize,
 {
     type In = S::In;
-    type Out = P::Out;
+    type Out = Option<P::Out>;
```

Comment:
I'm not super happy with this, but it's the first thing I thought of to deal with `Parse::done` clearly having `None` as an acceptable outcome.